web/ui/react-app/src/pages/alerts/AlertContents.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { FC, useState, Fragment } from 'react';
+import { ButtonGroup, Button, Row, Badge } from 'reactstrap';
+import CollapsibleAlertPanel from './CollapsibleAlertPanel';
+import Checkbox from '../../Checkbox';
+import { isPresent } from '../../utils/func';
+
+export type RuleState = keyof RuleStatus<any>;
+
+export interface Rule {
+  alerts: Alert[];
+  annotations: Record<string, string>;
+  duration: number;
+  health: string;
+  labels: Record<string, string>;
+  name: string;
+  query: string;
+  state: RuleState;
+  type: string;
+}
+
+export interface RuleStatus<T> {
+  firing: T;
+  pending: T;
+  inactive: T;
+}
+
+export interface AlertsProps {
+  groups?: RuleGroup[];
+  statsCount: RuleStatus<number>;
+}
+
+interface Alert {
+  labels: Record<string, string>;
+  state: RuleState;
+  value: string;
+  annotations: Record<string, string>;
+  activeAt: string;
+}
+
+interface RuleGroup {
+  name: string;
+  file: string;
+  rules: Rule[];
+  interval: number;
+}
+
+const AlertsContent: FC<AlertsProps> = ({ groups = [], statsCount }) => {
+  const [state, setState] = useState<RuleStatus<boolean>>({
+    firing: true,
+    pending: true,
+    inactive: true,
+  });
+  const [showAnnotations, setShowAnnotations] = useState(false);
+
+  const toggle = (ruleState: RuleState) => () => {
+    setState({
+      ...state,
+      [ruleState]: !state[ruleState],
+    });
+  };
+
+  return (
+    <>
+      <ButtonGroup className="mb-3">
+        <Button active={state.inactive} onClick={toggle('inactive')} color="primary">
+          Inactive({statsCount.inactive})
+        </Button>
+        <Button active={state.pending} onClick={toggle('pending')} color="primary">
+          Pending({statsCount.pending})
+        </Button>
+        <Button active={state.firing} onClick={toggle('firing')} color="primary">
+          Firing({statsCount.firing})
+        </Button>
+      </ButtonGroup>
+      <Row className="mb-2">
+        <Checkbox
+          id="show_annotations"
+          wrapperStyles={{ margin: '0 0 0 15px', alignSelf: 'center' }}
+          onClick={() => setShowAnnotations(!showAnnotations)}
+        >
+          Show annotations
+        </Checkbox>
+      </Row>
+      {groups.map((group, i) => {
+        return (
+          <Fragment key={i}>
+            <StatusBadges rules={group.rules}>
+              {group.file} > {group.name}
+            </StatusBadges>
+            {group.rules.map((rule, j) => {
+              return (
+                state[rule.state] && (
+                  <CollapsibleAlertPanel key={rule.name + i + j} showAnnotations={showAnnotations} rule={rule} />
```

Comment:
Is `i + j` actually needed here or can it just be `j` since `i` is the same within this list anyway?

Reply:
Yeah, you are right. I was looking for uniqueness but in this case, only ```j``` is enough